llvm-spirv/lib/SPIRV/SPIRVUtil.cpp

```diff
@@ -303,7 +303,7 @@ bool isSYCLBfloat16Type(llvm::Type *Ty) {
       return false;
     StringRef Name = ST->getName();
     Name.consume_front("class.");
-    if ((Name.startswith("cl::sycl::") ||
+    if ((Name.startswith("cl::sycl::") || Name.startswith("sycl::_V1::") ||
```

Comment:
This is mostly my ignorance. can you please outline why we used sycl::_V1 here and just sycl earlier? Thanks

Reply:
This PR makes two independent changes at once, only because decoupling them would be lots of useless work with little benefit.

First change is to get rid of `cl::` and put `sycl` directly into the global namespace. 

The second part is groundwork for possible future ABI breaking changes. The idea is that if we'd need, for example, make ABI breaking changes into `sycl::queue`, we'll do something like this:

```
// sycl.hpp:
namespace sycl {
inline namespace _V2 {
class queue {};   // That's what new code compiled with new compiler/RT would refer to by sycl::queue
}
}

// sycl_queue_v1_compat.cpp, linked into future libsycl.so
namespace sycl {
namespace _V1 {
class queue { ... };   // Old binaries linked with new sycl.so would use this.
// Old implementation
}
}
```

The exact machinery for that is to be defined later when/if we'd actually need it. However, we need to introduce `_V` right now because if our current `queue` would be `sycl::queue` instead of `sycl::_V1::queue` (that's how inline namespace is being mangled) we won't have any possibility to implement the sketchy solution above.

Reply:
Actually, I probably misunderstood the question. The proper fix upstream is using `^sycl::`, without `_V1` so that it would work if/when we'll start using `sycl::_V2` symbols.

Reply:
Thanks for the explanation Andrei. I understand now. I think this is good as long as it is documented somewhere.
I can approve the SPIRV changes. Thanks for this.

Reply:
Sorry. I had another minor concern. When Khronos repo and this repo both get pulled in, will it cause confusion to the pulldown co-ordinator about which version to keep? May be I am missing something?
Thanks

Reply:
I'm not sure I follow what problem llvm-spirv translator changes solves. What will happen when runtime will switch to sycl::_V2 symbols? Translator will leave bfloat16 types as user defined types instead of SPIR-V buillt-in types. Right?

In general, please, avoid making local changes to the translator. All changes should be committed to the Khronos repository. Ideally we should remove the copy of sources from our repository and fetch them directly from the Khronos repository.